### DISCUSSION TITLE
Lecture 6 问题: 将 e_i 转换成二进制表示

### DISCUSSION THREAD

**Discussion by luckyyang** [zkp-co-learning/zkp-co-learn#62] [Q&A（提问题在隔壁~）] [answered]
![2023-03-13 at 13 42](https://user-images.githubusercontent.com/1963646/224626739-a9449b61-b5e7-45db-9ac7-927dc19e95d6.png)

18'20'' 这里讲的是什么意思呢？没有看懂。链接 https://youtu.be/bTiNNw8lHpc?t=1100

**Comment by Howard-Hu** [comment, marked as answer]
# 1.  $e_i = \sum_{l=0}^{\lambda-1}e_{i,l}2^l$

这一步可以理解成是将$e_i$转化为2进制表示，其中$e_{i,l}$是第$l$位(从低位往高位数)的比特值；

如 $11=1 * 2^0 + 1 * 2^1 + 0 * 2^2 + 1 * 2^3$ 



# 2. $e_i = \sum_{j=0}^{s-1}\sum_{k=0}^{t-1}2^{j+s * k}e_{i,j+s * k}$ 

约定称 $e_{i,l}*2^l$ 为 $ e_i $的表示中的一项，这一步是将所有的 $ e_{i,l}2^l $ 写成放到 $ s*t $( $ s*t=\lambda $)的矩阵中，然后重排求和顺序，如 $ 11=1*2^0+1*2^1+0*2^2+1*2^3 $ ，我们将其项放到 $ 2*2 $的矩阵中
 $$
\begin{pmatrix}
1*2^0&1*2^1\\
0*2^2&1*2^3\\
\end{pmatrix}
 $$
然后逐列求和 $ 11=e_i=\sum_{j=0}^1\sum_{k=0}^1 2^{j+2*k}e_{i,j+2*k} $;其中 $ e_{i,0}=1,e_{i,1}=1,e_{i,2}=0,e_{i,3}=1 $

即求和顺序为 $ 11=(1*2^0+0*2^2)+(1*2^1+1*2^3) $



# 3.  $g_i^{e_i}=\prod \limits_{l-0}^{\lambda-1}g_i^{2^le_{i,l}}$

将 $ e_i = \sum_{l=0}^{\lambda-1}e_{i,l}2^l$代入$g_i^{e_i} $ 得 $ g_i^{\sum_{l=0}^{\lambda-1}2^le_{i,l}}=\prod \limits_{l-0}^{\lambda-1}g_i^{2^le_{i,l}} $

将 $ \prod \limits_{l-0}^{\lambda-1}g_i^{2^le_{i,l}} $ 按#2得方式重排得 $ \prod\limits_{j=0}^{s-1}\prod\limits_{k-0}^{t-1}g_i^{2^{j+s*k}e_i,j+s*k} $ 


# 附录
如果公式不渲染，看以下截图:
![image](https://user-images.githubusercontent.com/24404366/224637495-a4d4028f-4a47-482e-ba3d-0975b76a6291.png)
